### PR TITLE
GEOMESA-1228 Fixing CQL functions in queries

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

edit: this needs to stack on top of the stats PR - I'll wait until that is merged to rebase.

### CODE REVIEW

**Comment by cne1x** [comment]
Looks good to me.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreQueryTest.scala [hunk 1]
```diff
@@ -221,6 +221,33 @@ class AccumuloDataStoreQueryTest extends Specification with TestWithMultipleSfts
...
+          Seq(StrategyType.ATTRIBUTE, StrategyType.RECORD, StrategyType.Z2, StrategyType.Z3)
+
+      forall(positives) { query =>
+        val result = ds.getFeatureSource(sftName).getFeatures(query).features().toList
+        result must haveLength(1)
+        result.head.getID mustEqual "fid-1"
+      }
+      forall(negatives) { query =>
+        val result = ds.getFeatureSource(sftName).getFeatures(query).features().toList
+        result must beEmpty
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreQueryTest.scala]: Hmm...  could we trouble you for a fid-2 with a polygon;)?

File: geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/package.scala [hunk 2]
```diff
@@ -413,11 +413,7 @@ package object filter {
     (one, two) match {
       case (p: PropertyName, l: Literal) => Some(PropertyLiteral(p.getPropertyName, l, None, flipped = false))
       case (l: Literal, p: PropertyName) => Some(PropertyLiteral(p.getPropertyName, l, None, flipped = true))
-      case (_: PropertyName, _: PropertyName) | (_: Literal, _: Literal) => None
-      case _ =>
-        val msg = s"Unhandled expressions in strategy: ${one.getClass.getName}, ${two.getClass.getName}"
-        throw new RuntimeException(msg)
-
+      case _ => None
```
Comment (jnh5y) [inline geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/package.scala]: Nice.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreQueryTest.scala (hunk 1 above)
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreQueryTest.scala]: Sure - I'll add it when i rebase.

**Pull request merged**